dlpy/speech.py

```diff
@@ -0,0 +1,258 @@
+            self.data_path = self.data_path[:-1]
+        self.data_path += self.server_sep
+
+        server_type = get_cas_host_type(self.conn).lower()
+        is_server_unix = server_type.startswith("lin") or server_type.startswith("osx")
+
+        client_type = platform.system()
+        if (is_server_unix and client_type.startswith("Win")) or not (is_server_unix or client_type.startswith("Win")):
+            if local_path is None:
+                raise DLPyError("the \"local_path\" parameter is not specified. "
+                                "The CAS server and the Python client have different OS type (Windows/Linux), "
+                                "so please specify the \"local_path\" parameter.")
+            else:
+                self.local_path = local_path
+        else:
+            if local_path is None:
+                self.local_path = self.data_path
+                print("Note: the \"local_path\" parameter is not specified. "
+                      "The CAS server and the Python client have the same OS type (Windows/Linux), "
+                      "so simply use \"data_path\" as \"local_path\":", self.local_path)
+            else:
+                self.local_path = local_path
+
+        if not os.path.exists(self.local_path):
+            raise DLPyError("Invalid \"local_path\" value: does not exist.")
+        if not os.access(self.local_path, os.R_OK):
+            raise DLPyError("Invalid \"local_path\" value: does not have reading permission.")
+        if not os.access(self.local_path, os.W_OK):
+            raise DLPyError("Invalid \"local_path\" value: does not have writing permission.")
+
+        self.conn.loadactionset("audio", _messagelevel="error")
+        self.conn.loadactionset("deepLearn", _messagelevel="error")
+        self.conn.loadactionset("langModel", _messagelevel="error")
+
+        if acoustic_model_path is not None:
+            self.load_acoustic_model(acoustic_model_path)
+
+        if language_model_path is not None:
+            self.load_language_model(language_model_path)
+
+        self.data_caslib, self.data_path_after_caslib, _ = caslibify(self.conn, self.data_path, task="save")
+        self.data_caslib_path = self.conn.caslibinfo(caslib=self.data_caslib).CASLibInfo["Path"][0]
+        if not self.data_caslib_path.endswith(self.server_sep):
+            self.data_caslib_path += self.server_sep
+
+    def load_acoustic_model(self, acoustic_model_path):
+        """
+        Load the RNN acoustic model.
+
+        Parameters
+        ----------
+        acoustic_model_path: string
+            Specifies the absolute server-side path of the acoustic model file.
+            Please make sure the weights file and the weights attribute file are placed under the same directory.
+
+        """
+        self.acoustic_model = Model(self.conn)
+        self.acoustic_model.from_sashdat(self.conn, path=acoustic_model_path)
+
+        if self.acoustic_model is None:
```

Comment:
Shouldn't this test come before the `from_sashdat` line?  If self.acoustic_model is None, the previous line will blow up with an error about `from_sashdat` not being a method on `None`.

Reply:
I am not sure if this resolves the problem. When you create an instance of Model it cannot be None